src/management/Akka.Management/Cluster/Bootstrap/SelfAwareJoinDecider.cs

```diff
@@ -67,7 +68,8 @@ public bool MatchesSelf(ServiceDiscovery.ResolvedTarget target, (string host, in
         public bool HostMatches(string host, ServiceDiscovery.ResolvedTarget target)
         {
             var cleaned = _hostReplaceRegex.Replace(host, "");
-            return host == target.Host || (target.Address?.ToString() ?? "").Contains(cleaned);
+            return string.Equals(host, target.Host, StringComparison.OrdinalIgnoreCase) || 
```

Comment:
Key fix is here - DNS is case-insensitive by design and we need to treat it that way inside the join decider implementations.